react/components/BaseModal.tsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react'
+import { pick } from 'ramda'
+import Backdrop from './Backdrop'
+import Portal, { ContainerType } from './Portal'
+import { useResponsiveValue, ResponsiveInput } from 'vtex.responsive-values'
+
+interface Props extends React.DetailedHTMLProps<React.HTMLAttributes<HTMLDivElement>, HTMLDivElement> {
+  open: boolean
+  keepMounted?: boolean
+  container?: ContainerType
+  hideBackdrop?: boolean | ResponsiveInput<boolean>
+  backdropInvisible?: boolean | ResponsiveInput<boolean>
+  onBackdropClick?: (e: React.MouseEvent<HTMLDivElement>) => void
+}
+
+const styles: Record<string, React.CSSProperties> = {
+  container: {
+    position: 'fixed',
+    right: 0,
+    bottom: 0,
+    top: 0,
+    left: 0,
+    zIndex: 1300,
+  },
+} as const
+
+const BaseModal: React.FC<Props> = props => {
+  const {
+    children,
+    container,
+    open = true,
+    onBackdropClick,
+    keepMounted = false,
+    ...rest
+  } = props
+
+  const {
+    hideBackdrop = false,
+    backdropInvisible = false,
+  } = useResponsiveValue(pick(['hideBackdrop', 'backdropInvisible'], props))
+
+  const handleClick = (e: React.MouseEvent<HTMLDivElement>) => {
+    e.preventDefault()
+    e.stopPropagation()
```

Comment:
Why?

Reply:
This modal is usually inside of an element that has some other callbacks, like the summary that if you click it will go to the product's page, so I need to isolate this event to just the modal

Reply:
Does react bubbles the event up even with Portal? Did you test it?

Reply:
I had a bug a while ago about this hahaha. And there is in the docs explaining:
https://pt-br.reactjs.org/docs/portals.html#event-bubbling-through-portals